src/event.js

```diff
@@ -428,75 +398,40 @@ jQuery.event = {
 		return handlerQueue;
 	},
 
-	// Includes all common event props including KeyEvent and MouseEvent specific props
-	props: ( "altKey bubbles cancelable ctrlKey detail eventPhase " +
-		"metaKey shiftKey view which char charCode key keyCode " +
-		"button buttons clientX clientY offsetX offsetY pageX pageY screenX screenY toElement"
-	).split( " " ),
-
-	propHooks: {
-		which: function( event ) {
-			var button = event.button;
-
-			// Add which for key events
-			if ( event.which == null && rkeyEvent.test( event.type ) ) {
-				return event.charCode != null ? event.charCode : event.keyCode;
-			}
-
-			// Add which for click: 1 === left; 2 === middle; 3 === right
-			if ( !event.which && button !== undefined && rmouseEvent.test( event.type ) ) {
-				return ( button & 1 ? 1 : ( button & 2 ? 3 : ( button & 4 ? 2 : 0 ) ) );
-			}
-
-			return event.which;
-		},
-
-		pageX: function( event ) {
-			var eventDoc, doc, body;
-
-			// Calculate pageX if missing and clientX available
-			if ( event.pageX == null && event.clientX != null ) {
-				eventDoc = event.target.ownerDocument || document;
-				doc = eventDoc.documentElement;
-				body = eventDoc.body;
-
-				return event.clientX +
-					( doc && doc.scrollLeft || body && body.scrollLeft || 0 ) -
-					( doc && doc.clientLeft || body && body.clientLeft || 0 );
-			}
-
-			return event.pageX;
-		},
+	addProp: function( name, hook ) {
```

Comment:
I think supporting `fixHooks` and `props` in migrate should be simple if we go that route ([This snippet](https://github.com/jquery/jquery/pull/2860#issuecomment-215652293) is a potential migrate example).

We could also look for alternates to `addProp(name[, hook])` such as those I mentioned in a [previous comment](https://github.com/jquery/jquery/pull/2860#issuecomment-215304627). For example `jQuery.event.props[type] = true | getterMethod` would allow easier access to existing getters, but then we need to check for new `props` in `fix`.

That is the nice thing about `addProp(name[, hook])`. It does everything in this setup method, and nothing at event/fix time. This also allows creating separate and simpler getters for the hook vs no-hook cases.

Other ideas...
* Change `hook( this.originalEvent )` to `hook( this.originalEvent, wrappedHook || propGetter )` so overridden hooks have runtime access to the previous hook. This might require storing the hooks somewhere other then the closure of the getter.
* Add other params to to `hook` like the property `name`?